src/vaadin-overlay.html

```diff
@@ -671,10 +672,14 @@
       }
 
       /**
-       * Returns all attached overlays.
+       * Returns all attached overlays in visual stacking order.
        */
       static get __attachedInstances() {
-        return Array.from(document.body.children).filter(el => el instanceof OverlayElement);
+        return Array.from(document.body.children)
+          .filter(el => el instanceof OverlayElement)
+          .sort((a, b) => {
+            return getComputedStyle(a).zIndex - getComputedStyle(b).zIndex;
```

Comment:
If I'm not mistaken, this means that for every overlay (except the first one and the last one) `getComputedStyle` is called twice, right?

Not sure if it's worth truing to optimise, so not a blocker.